### DISCUSSION TITLE
Spout SDK and texture sharing to Kinc/Kha

### DISCUSSION THREAD

**Discussion by datee** [Kode/Kore#598] [Q&A]
Sorry if this is an obvious question - but im still trying to understand how things are pieced together :) 

Im trying to work with the Spout SDK to manage some texture sharing in my Kha/Kinc app.
The Spout receiver can get a shared texture from another application, but im unsure of how to handle this here. 

The code below is from an Openframeworks app - the Spout SDK needs the id and target to receive and update the texture in OpenGL.
Now, this is straight forward in Openframeworks - but how do i get access to the id/target (GLuint) of a texture here in Kinc?

```
SpoutReceiver.ReceiveTexture(myTexture.getTextureData().textureID, myTexture.getTextureData().textureTarget)
```

**Comment by RobDangerous** [comment]
The id is texture.impl.texture but I don't even know what a target is supposed to be. Is that some OpenGL jargon I don't know? Is that OpenFrameworks jargon? Is it maybe just a GL_FRAMEBUFFER?

**Comment by datee** [comment]
heh yeah, sorry about the OF jargon there :p Its just GLenum target, guess its not so relevant ..

so this seems to be a doable way of doing it then? :
`tex = new Kore::Graphics4::Texture(640, 360, Kore::Graphics4::Image::RGBA32, false); `
`receiver.ReceiveTexture(tex->kincTexture.impl.texture, GL_TEXTURE_2D);`

**Comment by RobDangerous** [comment]
I don't know, you tell me.

**Reply by datee** [reply]
yes - it works beautifully :)
